crates/telio-firewall/src/firewall.rs

```diff
@@ -275,7 +277,24 @@ impl Debug for IpConnWithPort {
             .field("remote_addr", &StdIpAddr::from(self.remote_addr))
             .field("remote_port", &self.remote_port)
             .field("local_addr", &StdIpAddr::from(self.local_addr))
-            .field("remote_addr", &self.local_port)
+            .field("local_port", &self.local_port)
+            .finish()
+    }
+}
+
+#[derive(Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
+struct IcmpConn {
+    remote_addr: IpAddr,
+    local_addr: IpAddr,
+    identifier: u16,
```

Comment:
I think identifier should be u32. ICMP identifier + seq number. Now if one ping process sends two ping requests they will have the same identifier but different seq number. In this implementation the second response will not get through, because there will be only one entry in the cache which will get removed by the first response.

Reply:
Fixed